tests/test_vcs_git/test_vcs_git.py

```diff
@@ -13,33 +11,50 @@ class GitTests(DxrInstanceTestCaseMakeFirst):
     def test_diff(self):
         """Make sure the diff link exists and goes to the right place."""
         response = self.client().get('/code/source/main.c')
-        ok_('/commit/cb339834998124cb8165aa35ed4635c51b6ac5c2" title="Diff" class="diff icon">Diff</a>' in response.data)
+        ok_('/commit/%s" title="Diff" class="diff icon">Diff</a>' % LATEST_REVISION in response.data)
 
     def test_blame(self):
         """Make sure the blame link exists and goes to the right place."""
         response = self.client().get('/code/source/main.c')
-        ok_('/blame/cb339834998124cb8165aa35ed4635c51b6ac5c2/main.c" title="Blame" class="blame icon">Blame</a>' in response.data)
+        ok_('/blame/%s/main.c" title="Blame" class="blame icon">Blame</a>' % LATEST_REVISION in response.data)
 
     def test_raw(self):
         """Make sure the raw link exists and goes to the right place."""
         response = self.client().get('/code/source/main.c')
-        ok_('/raw/cb339834998124cb8165aa35ed4635c51b6ac5c2/main.c" title="Raw" class="raw icon">Raw</a>' in response.data)
+        ok_('/raw/%s/main.c" title="Raw" class="raw icon">Raw</a>' % LATEST_REVISION in response.data)
 
     def test_log(self):
         """Make sure the log link exists and goes to the right place."""
         response = self.client().get('/code/source/main.c')
-        ok_('/commits/cb339834998124cb8165aa35ed4635c51b6ac5c2/main.c" title="Log" class="log icon">Log</a>' in response.data)
+        ok_('/commits/%s/main.c" title="Log" class="log icon">Log</a>' % LATEST_REVISION in response.data)
 
     def test_permalink(self):
         """Make sure the permalink link exists and goes to the right place."""
         response = self.client().get('/code/source/main.c')
-        ok_('/rev/cb339834998124cb8165aa35ed4635c51b6ac5c2/main.c" title="Permalink" class="permalink icon">Permalink</a>' in response.data)
+        ok_('/rev/%s/main.c" title="Permalink" class="permalink icon">Permalink</a>' % LATEST_REVISION in response.data)
+        # Test that it works for this revision and the last one.
+        response = self.client().get('/code/rev/%s/main.c' % LATEST_REVISION)
+        eq_(response.status_code, 200)
+        response = self.client().get('/code/rev/%s/main.c' % PREVIOUS_REVISION)
+        eq_(response.status_code, 200)
+
+    def test_deep_permalink(self):
+        """Make sure the permalink link exists and goes to the right place for files not in the
+        top-level directory. This test makes sure that the permalink works even for files not in
```

Comment:
Looks good. I think we can remove "This test makes sure that the permalink works even for files not in the top level git root directory, since" without any loss of meaning. (And if you're going to skip a line before the function body, do it inside the quotes; that's the convention we mostly follow.)